src/Engine/ProtoCore/FFI/CLRObjectMarshaler.cs

```diff
@@ -612,6 +612,9 @@ public override object UnMarshal(StackValue dsObject, ProtoCore.Runtime.Context
     class CLRObjectMarshaler : FFIObjectMarshaler
     {
         private static readonly Dictionary<Type, FFIObjectMarshaler> mPrimitiveMarshalers;
+        private static Type mCachedObjType;
```

Comment:
Minor but I would make these non-static members so that they can immediately be GC'd after the singleton goes out of scope.